src/Markdig/Renderers/RendererBase.cs

```diff
@@ -16,21 +16,34 @@ namespace Markdig.Renderers
     /// <seealso cref="IMarkdownRenderer" />
     public abstract class RendererBase : IMarkdownRenderer
     {
-        private readonly Dictionary<Type, IMarkdownObjectRenderer> renderersPerType;
-        private IMarkdownObjectRenderer? previousRenderer;
-        private Type? previousObjectType;
-        internal int childrenDepth = 0;
+        private readonly Dictionary<RuntimeTypeHandle, IMarkdownObjectRenderer?> _renderersPerType = new();
+        internal int _childrenDepth = 0;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="RendererBase"/> class.
         /// </summary>
-        protected RendererBase()
+        protected RendererBase() { }
+
+        private IMarkdownObjectRenderer? GetRendererInstance(MarkdownObject obj)
         {
-            ObjectRenderers = new ObjectRendererCollection();
-            renderersPerType = new Dictionary<Type, IMarkdownObjectRenderer>();
+            RuntimeTypeHandle typeHandle = Type.GetTypeHandle(obj);
+            Type objectType = obj.GetType();
+
+            for (int i = 0; i < ObjectRenderers.Count; i++)
+            {
+                var renderer = ObjectRenderers[i];
+                if (renderer.Accept(this, objectType))
+                {
+                    _renderersPerType[typeHandle] = renderer;
+                    return renderer;
+                }
+            }
+
+            _renderersPerType[typeHandle] = null;
+            return null;
```

Comment:
We weren't marking "unknown" types before, meaning that if a type didn't have an associated renderer, we would loop through the list every time.

I.e. for every `LinkReferenceDefinition`, `ContainerInline`, `ContainerBlock`, etc. we would fallback from `Dictionary` to looping through all renderers.

I guess is that this is why the `Type.IsAssignableFrom` was so much more expensive compared to `is`.